khmer/_oxli/graphs.pyx

```diff
@@ -1,20 +1,261 @@
+        trimmed_at = deref(self.c_table).trim_on_abundance(data, abundance)
+        return sequence[:trimmed_at], trimmed_at
+
+    def trim_below_abundance(self, sequence, abundance):
+        """Trim sequence at first k-mer above the given abundance."""
+        data = self._valid_sequence(sequence)
+        trimmed_at = deref(self.c_table).trim_below_abundance(data, abundance)
+        return sequence[:trimmed_at], trimmed_at
+
+    def find_spectral_error_positions(self, sequence, max_count):
+        """Identify positions of low-abundance k-mers."""
+        data = self._valid_sequence(sequence)
+        posns = (deref(self.c_table).find_spectral_error_positions(data,
+                                                                   max_count))
+        return posns
+
+    def consume_seqfile_with_reads_parser(self, read_parser):
+       """Count all k-mers from read_parser."""
+       cdef unsigned long long n_consumed = 0
+       cdef unsigned int total_reads = 0
+
+       cdef CPyReadParser_Object* parser = <CPyReadParser_Object*>read_parser
+
+       deref(self.c_table).consume_seqfile[CpFastxReader](parser.parser,
+                                                          total_reads,
+                                                          n_consumed)
+
+    def consume_seqfile(self, file_name):
+        """Count all k-mers from file_name."""
+        cdef unsigned long long n_consumed = 0
+        cdef unsigned int total_reads = 0
+
+        read_parser = ReadParser(file_name)
+        cdef CPyReadParser_Object* parser = <CPyReadParser_Object*>read_parser
+        deref(self.c_table).consume_seqfile[CpFastxReader](parser.parser,
+                                                           total_reads,
+                                                           n_consumed)
+
+    def abundance_distribution(self, file_name, tracking):
+        """Calculate the k-mer abundance distribution over reads in file_name."""
+        cdef CPyReadParser_Object* parser
+        if isinstance(file_name, str):
+            read_parser = ReadParser(file_name)
+            parser = <CPyReadParser_Object*>read_parser
+
+        else:
+            raise ValueError('Expected file_name to be string, '
+                             'got {} instead.'.format(type(file_name)))
+
+        cdef CPyHashtable_Object* hashtable
+        if isinstance(tracking, (Nodetable, Nodegraph)):
+            hashtable = <CPyHashtable_Object*>tracking
+        else:
+            raise ValueError('Expected `tracking` to be a Nodetable or '
+                             'Nodegraph, got {} instead.'.format(type(tracking)))
+
+        cdef uint64_t * x = deref(self.c_table).abundance_distribution[CpFastxReader](
+                parser.parser, hashtable.hashtable)
+        abunds = []
+        for i in range(MAX_BIGCOUNT):
+            abunds.append(x[i])
+        return abunds
+
+    def abundance_distribution_with_reads_parser(self, read_parser, tracking):
```

Comment:
Trying to remember why exactly I went with this. Part of this is there are a bunch of tests that start failing when I add:
```python
from khmer._oxli.parsing import FastxParser as ReadParser
```
to khmer/__init__.py, which would seem to be a useful thing to do to keep all the people using `ReadParser`s happy. Not sure this is super tricky, but can't work it out atm :-/ Merge now, improve later?